dockstore-webservice/src/main/java/io/dockstore/webservice/resources/WorkflowResource.java

```diff
@@ -803,8 +804,16 @@ public Workflow publish(@ApiParam(hidden = true) @Parameter(hidden = true, name
     public Workflow getWorkflowByPath(@ApiParam(hidden = true) @Parameter(hidden = true, name = "user")@Auth User user,
             @Parameter(name = "repository", description = "Repository path", required = true, in = ParameterIn.PATH) @ApiParam(value = "repository path", required = true) @PathParam("repository") String path,
             @Parameter(name = "include", description = "Comma-delimited list of fields to include: " + VALIDATIONS + ", " + ALIASES, in = ParameterIn.QUERY) @ApiParam(value = "Comma-delimited list of fields to include: " + VALIDATIONS + ", " + ALIASES) @QueryParam("include") String include,
-            @Parameter(name = "subclass", description = "Which Workflow subclass to retrieve. One of the folowwing: " + SERVICE + ", " + BIOWORKFLOW +  ", " + APPTOOL, in = ParameterIn.QUERY, required = true) @ApiParam(value = "Which Workflow subclass to retrieve. One of the following: " + SERVICE + ", " + BIOWORKFLOW +  ", " + APPTOOL) @QueryParam("subclass") String subclass) {
-        final Class<? extends Workflow> targetClass = getSubClass(subclass);
+            @Parameter(name = "subclass", description = "Which Workflow subclass to retrieve. One of the folowing: " + SERVICE + ", " + BIOWORKFLOW +  ", " + APPTOOL, in = ParameterIn.QUERY, required = true) @ApiParam(value = "Which Workflow subclass to retrieve. One of the following: " + SERVICE + ", " + BIOWORKFLOW +  ", " + APPTOOL) @QueryParam("subclass") String subclass,
+            @Parameter(name = "services", description = "Should only be used by Dockstore CLI versions < 1.11.0. Indicates whether to get a service or workflow", in = ParameterIn.QUERY, hidden = true) @ApiParam(value = "services", defaultValue = "false") @QueryParam("services") boolean services, @Context ContainerRequestContext containerContext) {
+        final Class<? extends Workflow> targetClass;
+        List<String> userAgent = containerContext.getHeaders().get("User-Agent");
+        if (userAgent != null && userAgent.get(0).contains("Dockstore-CLI")) {
```

Comment:
In my previous comment, I suggest you don't need to check for the CLI at all, but if I'm wrong or if you approach it differently: 

If you don't check the version, then this condition will be true for the 1.12, 1.13, etc. CLI, and I don't think you want that.